x/wasm/types/validation.go

```diff
@@ -40,6 +41,9 @@ func ValidateLabel(label string) error {
 	if len(label) > MaxLabelSize {
 		return ErrLimit.Wrapf("cannot be longer than %d characters", MaxLabelSize)
 	}
+	if label != strings.TrimSpace(label) {
+		return ErrInvalid.Wrapf("'%s'This label is not trimmed", label)
```

Comment:
Also very good to reject all labels with whitespaces as default behaviour. This ensures a clean state.
No need to return the original string, just the error context. For example:
```suggestion
		return ErrInvalid.Wrap("label must not start/end with whitespaces")
```